BuildProject/tests/App.test.js

```diff
@@ -0,0 +1,37 @@
+import wd from 'wd';
+
+jasmine.DEFAULT_TIMEOUT_INTERVAL = 600000;
+const PORT = 4723;
+
+const config = {
+  platformName: "iOS",
+  platformVersion: "14.4",
+  deviceName: "iPhone 11",
+  app: "build/BuildProject.ipa",
+  automationName: "XCUITest",
+};
+
+const driver = wd.promiseChainRemote('localhost', PORT);
+
+beforeAll(async () => {
+  await driver.init(config);
+})
+
+test('Test All Promoted Logging Calls', async () => {
+  const testAllButton = await driver.elementByAccessibilityId('test-all-button');
+  const messagesText = await driver.elementByAccessibilityId('messages-text');
+  const tries = 10;
+  var i;
+  for (i = 0; i < tries; i++) {
```

Comment:
nit - Haha, got it.  I'd hope wd had a way to handle this retry structure automatically.

If you think you'd do this a bunch... I found this open source project.  MIT licensed.  You can probably just copy the code.
https://github.com/juliemr/webdriverjs-retry

not a blocker.

Reply:
I looked at this, and unfortunately this library doesn't apply because Webdriver isn't throwing an error or timing out, it's just not populating the text box string. There's no way (AFAICT) to tell the retry library to retry on a custom condition, it requires a Webdriver error or timeout.

Reply:
Had to rewrite the retry logic because that dependency brings in the entire selenium-webdriver dependency.